From 9bf5efa25589736de6d47e62299ad11083451eed Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Tue, 5 Nov 2024 16:40:32 +0530
Subject: [PATCH] wifi:ath11k: Handle duplicate descriptor case in
 mon dest ring

When the software cookie from the MSDU link descriptor matches the
previous software cookie in the monitor destination ring, the physical
address for the corresponding ring is not stored. This results in invalid
memory access when a duplicate entry is detected in the monitor
destination ring. Fix it by caching the last physical address and
comparing it with the new physical address. Skip the monitor destination
ring processing if they are equal.

Patch-dependency: none

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/dp_rx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath11k/dp_rx.c b/drivers/net/wireless/ath/ath11k/dp_rx.c
index 5fb12eeb..c6fa9506 100644
--- a/drivers/net/wireless/ath/ath11k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath11k/dp_rx.c
@@ -5949,6 +5949,7 @@ ath11k_dp_rx_mon_mpdu_pop(struct ath11k *ar, int mac_id,
 					   i, pmon->mon_last_buf_cookie);
 				drop_mpdu = true;
 				pmon->rx_mon_stats.dup_mon_buf_cnt++;
+				pmon->mon_last_linkdesc_paddr = paddr;
 				continue;
 			}
 			buf_id = FIELD_GET(DP_RXDMA_BUF_COOKIE_BUF_ID,
-- 
2.34.1

